### PR TITLE
MSC2610: Remove `m.login.oauth2` User-Interactive Authentication type from the specification

### PR DESCRIPTION
**Pull request by richvdh**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/rav/proposal/remove-oauth2-auth-type/proposals/2610-remove-oauth2-auth-type.md)

### CODE REVIEW

**Comment by janrenz** [comment]
Ist this implemeted in Synapse by https://github.com/matrix-org/synapse/pull/7256

**Comment by richvdh** [comment]
> Ist this implemented in Synapse by matrix-org/synapse#7256

no, that uses `m.login.sso`

File: proposals/2610-remove-oauth2-auth-type.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+real use. This MSC proposes removing them.
+
+## Proposal
+
+The definition of
+[OAuth2-based](https://matrix.org/docs/spec/client_server/r0.6.1#oauth2-based)
+authentication is incomplete. [OAuth2](https://oauth.net/2/) is best considered
+as a framework for implementing authentication protocols rather than a protocol
+in its own right, and this section says nothing about the grant types, flows
+and scopes which a compliant implemenation should understand.
```
Comment (anoadragon453) [inline proposals/2610-remove-oauth2-auth-type.md:18]:
```suggestion
and scopes which a compliant implementation should understand.
```

File: proposals/2610-remove-oauth2-auth-type.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+[OpenID Connect](https://openid.net/connect/), but this has already been
+implemented in Matrix via the `m.login.sso` authentication type.
+
+The `m.login.oauth2` section is therefore unimplementable in its current form,
+and redundant. It should be removed from the specification to reduce confusion.
+
+## Alternatives
+
+It would be possible to extend the definition so that it is complete: as
+mentioned above, a likely implemenation would be based on OpenID
```
Comment (anoadragon453) [inline proposals/2610-remove-oauth2-auth-type.md:30]:
```suggestion
mentioned above, a likely implementation would be based on OpenID
```

**Comment by richvdh** [comment]
there's no implementation work blocking this, so:

@mscbot fcp merge

**Comment by ara4n** [comment]
are we aware of anyone using m.login.oauth2 in the wild who is going to be bitten by this?

**Comment by richvdh** [comment]
> are we aware of anyone using m.login.oauth2 in the wild who is going to be bitten by this?

I'm not aware of any implementations of it. Indeed, I don't see how it *could* be implemented, given the current state of the spec: there simply isn't enough information there.

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**